_posts/2019-07-29-plonk-stack.md

```diff
@@ -0,0 +1,211 @@
+---
+title: "Introducing the PLONK Stack for Cloud Native Developers"
+description: You've heard of LAMP, JAM, and MEAN, but what's the PLONK stack, and why should you be considering it for your Cloud Native Applications?
+date: 2019-08-04
+image: /images/2019-intro-plonk/map-reader.jpg
+categories:
+  - community
+  - end-user
+  - production
+  - kubernetes
+  - plonk
+  - cloud native
+author_staff_member: alex
+dark_background: true
+---
+
+You've heard of LAMP, JAM, and MEAN, but what's the PLONK stack, and why should you be considering it for your Cloud Native Applications?
+
+<blockquote class="twitter-tweet"><p lang="en" dir="ltr">Introducing the PLONK! stack.<br><br>Probably fits in less than 1GB of RAM when used with k3s. <a href="https://twitter.com/PrometheusIO?ref_src=twsrc%5Etfw">@PrometheusIO</a> <a href="https://twitter.com/linkerd?ref_src=twsrc%5Etfw">@linkerd</a> <a href="https://twitter.com/openfaas?ref_src=twsrc%5Etfw">@openfaas</a> <a href="https://twitter.com/nats_io?ref_src=twsrc%5Etfw">@nats_io</a> <a href="https://twitter.com/kubernetesio?ref_src=twsrc%5Etfw">@kubernetesio</a> <a href="https://twitter.com/CloudNativeFdn?ref_src=twsrc%5Etfw">@CloudNativeFdn</a> <a href="https://twitter.com/Rancher_Labs?ref_src=twsrc%5Etfw">@Rancher_Labs</a> <a href="https://t.co/TMTtu2oRup">pic.twitter.com/TMTtu2oRup</a></p>&mdash; Alex Ellis (@alexellisuk) <a href="https://twitter.com/alexellisuk/status/1154385264925650950?ref_src=twsrc%5Etfw">July 25, 2019</a></blockquote> <script async src="https://platform.twitter.com/widgets.js" charset="utf-8"></script>
+
+The PLONK stack combines the following Cloud Native technologies from the Cloud Native Landscape:
+
+* Prometheus
+* Linkerd
+* OpenFaaS
+* NATS
+* Kubernetes
+
+Out of the projects listed, four are CNCF-hosted. Of the CNFC projects, 2 are "graduated" and the other two are "incubating" and showing signs of graduation soon. In this post I'll explain what each of these CNCF projects lends to OpenFaaS, and how they can build a first-class FaaS or PaaS when combined.
```

Comment:
On its own OpenFaaS isnt a first-class FaaS?

Reply:
I think I can see where you are coming from. Do you think OpenFaaS is a first class FaaS without clustering like K for Kubernetes? How else would you write this sentence?

Reply:
I read it from a 3rd person's perspective.  I'm highlighting the inference that someone reading this with innocent eyes might draw.  The answer lies, I suspect, in the reason why you'd use PLONK.  What are you getting above and beyond using OF on its own (arguably if you're using K8s you're only actually the L short)

Reply:
Yes I am not sure what to do about that.

OpenFaaS is P O N K - L makes it sound better.

Reply:
I've rewritten this. I hope it's better.